### DISCUSSION TITLE
[auth][error] TypeError: fetch failed

### DISCUSSION THREAD

**Discussion by nvijayap** [nodejs/undici#2638] [Q&A]
Any idea why I am facing this (when using `@auth/core` in node v21 environment) ? ...
```
[auth][error] TypeError: fetch failed
    at node:internal/deps/undici/undici:12442:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async getAuthorizationUrl (file:///Users/xxx/pnd4/.output/server/node_modules/@auth/core/lib/actions/signin/authorization-url.js:18:35)
```
Is there a way to bypass `undici` altogether and use alternative implementations of `fetch` (say by setting an env var)?

**Comment by metcoder95** [comment]
👋 
At first glance, it seems an issue with retrieving some resources that `@auth/core` is trying to fetch. I suggest opening an issue with them to seek some support.

**Reply by nvijayap** [reply]
Ok, per your suggestion, I have opened https://github.com/nextauthjs/next-auth/issues/9811
